### ISSUE TITLE
BUG: Slow start-up time (regression?)

### ISSUE THREAD

**Issue by amperag** [zsviczian/obsidian-excalidraw-plugin#2124]
### Have you searched for existing issues (including closed ones)?

- [X] Yes, I have searched the existing issues.

### Does this bug persist in a new vault with only Excalidraw installed?

- [X] Yes, I have verified the issue persists.

### Your environment

SYSTEM INFO:
	Obsidian version: v1.7.7
	Installer version: v1.7.5
	Operating system: Windows 10 Pro 10.0.19045
	Login status: not logged in
	Language: de
	Insider build toggle: off
	Live preview: on
	Base theme: light
	Community theme: Border v1.10.9
	Snippets enabled: 3
	Restricted mode: off
	Plugins installed: 10
	Plugins enabled: 10
		1: Dataview v0.5.67
		2: Auto Note Mover v1.2.0
		3: Excalidraw v2.6.7
		4: Sort & Permute lines v0.7.0
		5: Recent Files v1.7.2
		6: Paste image rename v1.6.1
		7: Omnisearch v1.25.1
		8: Text Extractor v0.5.2
		9: Tag Wrangler v0.6.1
		10: Zotero Integration v3.2.1

### Describe the bug

To me it seems that since the last couple of minor releases since 2.6.2 the Excalidraw plugin is again really slow at start-up.

The early 2.6.x releases (<= 2.6.2) provided a massive performance gain at start-up. Excalidraw was superfast! See my comment here: https://github.com/zsviczian/obsidian-excalidraw-plugin/issues/2068#issuecomment-2443518118

Yet, since the last 2.6.x releases (>2.6.2) at least for me the start-up has become really slow. See "Additional context".

### Steps to reproduce

1. Open Obsidian
2. Start-up time of at least 3.000 ms.

### Expected behavior

Start-up time of below 500 ms.

### Additional context

Obsidian start-up time breakdown

Obsidian version: v1.7.7
Installer version: v1.7.5
Operating system: Windows 10 Pro 10.0.19045

- Total startup time: 12.524ms
- Initialization: 562ms
- Vault (2.736 files): 893ms
- Workspace (10 tabs, 8 deferred): 981ms
- Core plugins: 417ms
- Community plugins (10 active): 9.671ms
  - Excalidraw (v2.6.7): 5.413ms
  - Text Extractor (v0.5.2): 1.585ms
  - Zotero Integration (v3.2.1): 778ms
  - Omnisearch (v1.25.1): 518ms
  - Dataview (v0.5.67): 409ms
  - Auto Note Mover (v1.2.0): 289ms
  - Paste image rename (v1.6.1): 110ms
  - Tag Wrangler (v0.6.1): 91ms
  - Recent Files (v1.7.2): 73ms
  - Sort & Permute lines (v0.7.0): 61ms

**Comment by zsviczian** [comment]
Here's my cold startup:

Obsidian start-up time breakdown

Obsidian version: v1.7.7
Installer version: v1.7.5
Operating system: Windows 11 Home 10.0.22631

- Total startup time: 2,038ms
- Initialization: 409ms
- Vault (16,992 files): 914ms
- Workspace (18 tabs, 16 deferred): 148ms
- Core plugins: 11ms
- Community plugins (18 active): 557ms
  - Excalidraw (v2.6.7): 164ms
  - VSCode Editor (v1.0.4): 139ms
  - Map View (v5.0.3): 79ms
  - PDF++ (v0.40.13): 45ms
  - Dataview (v0.5.67): 27ms
  - ExcaliBrain (v0.2.15): 15ms
  - Templater (v2.9.1): 11ms
  - Jump-to-Date (v1.0.18): 10ms
  - Style Settings (v1.0.9): 8ms
  - Note Refactor (v1.8.2): 8ms
  - Readwise Official (v2.1.4): 7ms
  - AI for Templater (v1.0.11): 5ms
  - Text Transporter (v1.0.12): 3ms
  - Pane Relief (v0.5.6): 3ms
  - BRAT (v1.0.5): 2ms
  - Recent Files (v1.7.1): 2ms
  - Regex Find/Replace (v1.2.0): 1ms
  - Reveal Active File Button (v2.0.3): 1ms

**Comment by zsviczian** [comment]
@amperag please install 2.6.8-beta-1
After loading, please open the developer console (CTRL+SHIFT+i)
and run the following commands:
```js
ExcalidrawAutomate.printStartupBreakdown();
```

Then copy the result in here.

**Comment by zsviczian** [comment]
The times under "Workspace onLayoutReady" happen after Obsidian workspace is ready, thus they do not impact startup time measured by Obsidian. They are included only for additional optimization, but have no relevance to the startup time you see in the report.
The 97ms is the startup time of Excalidraw that I have control over. The rest is the time for the java engine to read in and parse the plugin main.js file. I'm assuming Obsidian runs a number of startup tasks in parallel, thus the load on the PC impacts the java parser reading main.js.
- Based on testing I sense that the number of tabs open in the workspace seems to have an affect on the startup time. Try restarting with all tabs closed.
- Please create an empty vault, install only Excalidraw and test startup time. This will give you the actual time for Excalidraw, the rest is due to the size of the Vault and other plugins being initialized.
- The ultimate solution would be for Obsidian to support deploying asset files with plugins, not just a single main.js file. But this is not supported, thus maintaining Excalidraw 100% local, I have no other option, but to include everything in the main.js file.

**Comment by zsviczian** [comment]
Sorry, I don't have a take on all this testing. unpacking and loading mermaid happen after the workspace have loaded, so the number you see has no relevance to Obsidian startup.
Startup times between your individual experiments are anecdotal measurements and likely differ due to other load on your system and not due to Excalidraw.
I propose to close this issue here. I don't plan to spend more effort on this as the solution is outside my control.

**Comment by Ben9986** [comment]
@amperag i was having this issue and updating the installer version to 1.7.7 seems to have fixed it for now. Re download the latest exe installer from obsidian and see if it helps.

**Comment by zsviczian** [comment]
I doubt this is an Obsidian installer issue. 
This could also be related to system RAM. If you have limited RAM 8GB? maybe even 16GB? then if you have many apps or many tabs in the browser, then Windows might be swapping to disk causing slow processing. One way to test this is to restart Windows, wait until the desktop loads fully (give it 1-2 minutes to be sure), then start Obsidian before any other app to see if times are better.

**Comment by zsviczian** [comment]
@amperag, maybe this is the right time to grab a Macbook M1...? I think it is a price/value no-brainer (if you can still get it). I bought mine 2 years ago so I can test Excalidraw on MacOS given there is a large Mac user base. M1 Macbook runs all my apps nicely. Before this purchase, I only had Windows and Linux.... and for many tasks I still prefer Windows, but once you get over the MacOS quirks, as a general-purpose note-taking, on-the-go laptop, I am pretty happy with my M1.

**Comment by Ben9986** [comment]
I doubt its a ram issue either. I have 16GB and with nothing else running it still occurred. Seems to have gone away when i updated the installer. Not necessarily saying its the cause but certainly a correlation.

**Comment by JosephBrooksbank** [comment]
FWIW, I have a rather overspecced laptop (64GB ram, i9) and my obsidian takes ~5 seconds to start up the first time. I believe mostly for excalidraw. I don't know how to force a 'cold' start to test with `ExcalidrawAutomate.printStartupBreakdown();`, further startups are near instant

**Comment by zsviczian** [comment]
I understand your concern, and I want to be very clear about this. Please stop adding to this list regarding startup time. If you want faster plugin startup, I recommend engaging with the Obsidian developer team to advocate for features like allowing plugins to deploy assets. On my end, I have done everything within my power.

After extensive measurement and investigation, I’ve found that I can influence only about 0.2% of the total startup time. The majority of the delay comes from Obsidian loading the file, the Electron browser parsing it, and the Excalidraw CSS being integrated into the Obsidian DOM.

If this remains a dealbreaker for you, I completely understand. However, continuing to pursue this—now with 26 comments on a closed issue—will only lead to unnecessary frustration on my side. I hope this clarifies the situation. Thank you for understanding.

<img width="1160" alt="image" src="https://github.com/user-attachments/assets/9f56d3a5-8cc6-4f7a-971b-3c36c2737e8c" />
